### ISSUE TITLE
Most recent requirements change results in pip install error

### ISSUE THREAD

**Issue by bschreck** [m-bain/whisperX#602]
The change to depending on git repo of faster-whisper instead of pypi produces an error. Pip installing from latest commit results in:

```Cloning https://github.com/m-bain/whisperx.git to /tmp/pip-req-build-af_jhve1
7.586   Running command git clone --filter=blob:none --quiet https://github.com/m-bain/whisperx.git /tmp/pip-req-build-af_jhve1
10.04   Resolved https://github.com/m-bain/whisperx.git to commit 5a16e59217b96938460f011ddd8f1591b0de8afe
10.09   Preparing metadata (setup.py): started
11.12   Preparing metadata (setup.py): finished with status 'error'
11.16   error: subprocess-exited-with-error
11.16   
11.16   × python setup.py egg_info did not run successfully.
11.16   │ exit code: 1
11.16   ╰─> [41 lines of output]
11.16       /tmp/pip-req-build-af_jhve1/setup.py:4: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
11.16         import pkg_resources
11.16       Traceback (most recent call last):
11.16         File "/usr/local/lib/python3.11/dist-packages/pkg_resources/_vendor/packaging/requirements.py", line 35, in __init__
11.16           parsed = _parse_requirement(requirement_string)
11.16                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
11.16         File "/usr/local/lib/python3.11/dist-packages/pkg_resources/_vendor/packaging/_parser.py", line 64, in parse_requirement
11.16           return _parse_requirement(Tokenizer(source, rules=DEFAULT_RULES))
11.16                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
11.16         File "/usr/local/lib/python3.11/dist-packages/pkg_resources/_vendor/packaging/_parser.py", line 82, in _parse_requirement
11.16           url, specifier, marker = _parse_requirement_details(tokenizer)
11.16                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
11.16         File "/usr/local/lib/python3.11/dist-packages/pkg_resources/_vendor/packaging/_parser.py", line 126, in _parse_requirement_details
11.16           marker = _parse_requirement_marker(
11.16                    ^^^^^^^^^^^^^^^^^^^^^^^^^^
11.16         File "/usr/local/lib/python3.11/dist-packages/pkg_resources/_vendor/packaging/_parser.py", line 147, in _parse_requirement_marker
11.16           tokenizer.raise_syntax_error(
11.16         File "/usr/local/lib/python3.11/dist-packages/pkg_resources/_vendor/packaging/_tokenizer.py", line 165, in raise_syntax_error
11.16           raise ParserSyntaxError(
11.16       pkg_resources.extern.packaging._tokenizer.ParserSyntaxError: Expected end or semicolon (after name and no valid version specifier)
11.16           git+https://github.com/SYSTRAN/faster-whisper.git@0.10.0
11.16              ^
11.16       
11.16       The above exception was the direct cause of the following exception:
11.16       
11.16       Traceback (most recent call last):
11.16         File "<string>", line 2, in <module>
11.16         File "<pip-setuptools-caller>", line 34, in <module>
11.16         File "/tmp/pip-req-build-af_jhve1/setup.py", line 18, in <module>
11.16           install_requires=[
11.16                            ^
11.16         File "/tmp/pip-req-build-af_jhve1/setup.py", line 18, in <listcomp>
11.16           install_requires=[
11.16                            ^
11.16         File "/usr/local/lib/python3.11/dist-packages/pkg_resources/__init__.py", line 3171, in __init__
11.16           super(Requirement, self).__init__(requirement_string)
11.16         File "/usr/local/lib/python3.11/dist-packages/pkg_resources/_vendor/packaging/requirements.py", line 37, in __init__
11.16           raise InvalidRequirement(str(e)) from e
11.16       pkg_resources.extern.packaging.requirements.InvalidRequirement: Expected end or semicolon (after name and no valid version specifier)
11.16           git+https://github.com/SYSTRAN/faster-whisper.git@0.10.0
11.16              ^
11.16       [end of output]
```

**Comment by ffserro** [comment]
Same here:

Collecting git+https://github.com/m-bain/whisperx.git
  Cloning https://github.com/m-bain/whisperx.git to /tmp/pip-req-build-yqfcgnuv
  Running command git clone --filter=blob:none --quiet https://github.com/m-bain/whisperx.git /tmp/pip-req-build-yqfcgnuv
  Resolved https://github.com/m-bain/whisperx.git to commit 5a16e59217b96938460f011ddd8f1591b0de8afe
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  Preparing metadata (setup.py) ... error
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

**Comment by ultimatemoony** [comment]
![image](https://github.com/m-bain/whisperX/assets/122580973/ead2445c-0721-405d-b26a-896e8e8bcc13)
I have exactly the same error in Google Colab. I would love for it to work again just like before.

**Comment by SreemanthG** [comment]
Even I have the same issue

**Comment by rakuri255** [comment]
Sme here

**Comment by loganlebanoff** [comment]
If you want the newest changes, I got it working by changing the requirements.txt file from `git+https://github.com/SYSTRAN/faster-whisper.git@0.10.0` to `faster-whisper @ git+https://github.com/SYSTRAN/faster-whisper@master` and then running:
```sh
git clone https://github.com/m-bain/whisperx
cd whisperx
pip install -e .

# I had to install this because it was missing from requirements.txt I guess
pip install lightning-fabric
```

**Comment by sam-goodwin** [comment]
> git+https://github.com/SYSTRAN/faster-whisper@master and then running:

@loganlebanoff this didn't work for me. Same error

**Comment by avoss** [comment]
Same here. I solved it similar to [loganlebanof](https://github.com/loganlebanoff), but I changed the line in requirements.txt into `faster-whisper` and installed it manually:

```
pip install git+https://github.com/SYSTRAN/faster-whisper.git@0.10.0
cd whisperX && pip install -e .
```

**Comment by jim60105** [comment]
@bschreck please check if the issue has been resolved with #603
